.github/actions/setup-dfx/action.yml

```diff
@@ -0,0 +1,32 @@
+name: 'Setup DFX'
+description: Setup DFX
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/dfx
+          ~/.cache/dfinity
+        key: dfx-cache-0.11.1-2
+    # This step hangs on Github actions on Darwin for some reason, that
+    # is why we run this only on Linux for now
+    - name: Install DFX
+      shell: bash
+      env:
+        DFX_VERSION: 0.11.1
+      run: |
+        if [ -f ~/dfx/dfx ]
+        then
+          echo "DFX restored from cache"
+          ~/dfx/dfx --version
+        else
+          # sent env variable here so that '~' gets expanded
+          export DFX_INSTALL_ROOT=~/dfx
+          echo DFX not restored from cache, running install script:
+          echo DFX Version: "$DFX_VERSION"
+          echo DFX location: "$DFX_INSTALL_ROOT"
+          (yes || true) | sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
```

Comment:
guessing pipefail kicks in and fails everything? any idea why this wasn't the case before?

Reply:
Yes, exactly: https://github.com/dfinity/internet-identity/runs/8015828437?check_suite_focus=true

Apparently, the behavior changes when moved to a composite action.